android/src/main/java/io/sentry/react/RNSentryBreadcrumb.java

```diff
@@ -0,0 +1,87 @@
+package io.sentry.react;
+
+import com.facebook.react.bridge.ReadableMap;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+import io.sentry.Breadcrumb;
+import io.sentry.SentryLevel;
+
+public class RNSentryBreadcrumb {
+
+    @Nullable
+    public static String getCurrentScreenFrom(ReadableMap from) {
+        final @Nullable String maybeCategory = from.hasKey("category") ? from.getString("category") : null;
+        if (maybeCategory == null || !maybeCategory.equals("navigation")) {
+            return null;
+        }
+
+        final @Nullable ReadableMap maybeData = from.hasKey("data") ? from.getMap("data") : null;
+        if (maybeData == null) {
+            return null;
+        }
```

Comment:
suggestion: does the same thing at the end but we don't need to have a parameter maybe when we are sure we have the data after checking if the key exist.
```suggestion
        if (from.hasKey("data")) {
            return null;
        }
        final @Nullable ReadableMap data = from.getMap("data");
```